flink-core/src/main/java/org/apache/flink/api/common/state/AppendingState.java

```diff
@@ -60,8 +60,11 @@ public interface AppendingState<IN, OUT> extends State {
      *
      * <p>If null is passed in, the state value will remain unchanged.
      *
+     * <p>Except for list state, where null value is not allowed to be passed in.
```

Comment:
I think when referring to list state you mean the ListState interface. I do not think that a parent interface should document special behaviour for a child interface. This breaks to the solid principle Liskov Substitution Principle and is not good practice.

Reply:
Agreed. I believe it would be better to eliminate the description of the null value at the moment, considering the implementations have already become varied. We can consider refactoring the interfaces in the upcoming major version. What do you think? @davidradl